lib/ladybug/energy_model/schedule_ruleset_abridged.rb

```diff
@@ -0,0 +1,134 @@
+# may be used to endorse or promote products derived from this software without
+# specific prior written permission from the respective party.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE
+# UNITED STATES GOVERNMENT, OR THE UNITED STATES DEPARTMENT OF ENERGY, NOR ANY OF
+# THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+# OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *******************************************************************************
+
+require 'ladybug/energy_model/model_object'
+
+require 'json-schema'
+require 'json'
+require 'openstudio'
+
+module Ladybug
+  module EnergyModel
+    class ScheduleRulesetAbridged < ModelObject
+      attr_reader :errors, :warnings
+
+      def initialize(hash = {})
+        super(hash)
+
+        raise "Incorrect model type '#{@type}'" unless @type == 'ScheduleRulesetAbridged'
+      end
+
+      def defaults
+        result = {}
+        result
+      end
+
+      def find_existing_openstudio_object(openstudio_model)
+        object = openstudio_model.getScheduleRulesetByName(@hash[:name])
+        return object.get if object.is_initialized
+        nil
+      end
+
+      def create_openstudio_object(openstudio_model)
+        openstudio_schedule_ruleset = OpenStudio::Model::ScheduleRuleset.new(openstudio_model)
+        openstudio_schedule_ruleset.setName(@hash[:name])
+        
+        if @hash[:summer_designday_schedule]
+          summer_designday_schedule = OpenStudio::Model::ScheduleDay.new(openstudio_model)
+          values = @hash[:summer_designday_schedule][:values]
+          times = @hash[:summer_designday_schedule][:times]
+          values.each_index do |i|
+            summer_designday_schedule.addValue(OpenStudio::Time.new(0,times[i][0],times[i][1] ,0), values[i])
+          end 
+        end
+
+        openstudio_schedule_ruleset.setSummerDesignDaySchedule(summer_designday_schedule)
+
+        if @hash[:winter_designday_schedule]
+          winter_designday_schedule = OpenStudio::Model::ScheduleDay.new(openstudio_model)
+          values = @hash[:winter_designday_schedule][:values]
+          times = @hash[:winter_designday_schedule][:times]
+          values.each_index do |i|
+            winter_designday_schedule.addValue(OpenStudio::Time.new(0,times[i][0],times[i][1] ,0), values[i])
+          end 
```

Comment:
Given that you have these 5 lines repeated 3 times in this module, it might be a bit cleaner to just wrap them into their own hidden function and just pass the times and values to it.

Reply:
@chriswmackey I dont think I will be reuse the function for defaultDaySchedule since I don't create a new ScheduleDay for it but add values to schedule_ruleset.defaultDaySchedule. 
I have clubbed these lines in a new function for summer and winter design day however.